### PR TITLE
[@types/k6] Add type annotations for v0.40.0

### PR DESCRIPTION
**Pull request by oleiade**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/grafana/k6/pull/2630>
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by ppcano** [comment]
> This can potentially save days of time for you!

Love this bot. ;) I agree we should include tests on this PR.  There are tests for most of the packages.

Note that it is pending to upgrade the Typescript definition for the last two k6 versions. It is planned that  @nicolevanderhoeven work on them.

**Comment by oleiade** [comment]
Indeed, I didn't realize there was a test folder. Adding them now.

I'm uncertain if I understand why this is blocked on updating definitions for the previous two versions as this essentially an addition, and has zero impact on any other annotations outside the scope of the `k6/experimental/redis` module, though.

**Comment by ppcano** [comment]
Our current process is as follows:
- k6 releases a new version.
- k6 Typescript definitions are updated. The version of the k6 TS definition is upgraded to be in sync with the latest k6 version.
- Last, we upgrade other [project/s](https://github.com/grafana/k6-template-typescript) that use the Typescript definition.


Given the existing process (let's discuss this further), we should include this module when it is merged on `v0.40.0`. 

You suggest adding the TS definition before `k6/experimental/redis` is released in `v0.40.0`.  Why do you or other folks in the community need this?

The drawback is providing a TS module that is not available.

**Comment by oleiade** [comment]
There's been a misunderstanding here. I was not suggesting we merge this **before** `v0.40.0`'s release 👍🏻 

I understood "Note that it is pending to upgrade the Typescript definition for the last two k6 versions. It is planned that @nicolevanderhoeven work on them." as: it won't be merged hands in hands with the `v0.40.0` definition because some other work on the TS definitions need to happen before.

It's now clearer that it's not what you meant, nor what would happen 🙇🏻 

My understanding is now: we wait for `v0.40.0` to be released before we merge this. I'm fine with this, so if there was no further misunderstanding 😄 let's proceed with that indeed 👍🏻

**Comment by oleiade** [comment]
I've added tests, which the bot reports as passing 👍🏻

**Review by johnnyreilly — APPROVED** [review]

File: types/k6/experimental/timers.d.ts
Comment (mstoykov) [inline types/k6/experimental/timers.d.ts:3]:
```suggestion
```

I think this comment should probably be removed now that we are provinding them with the idea of people using them ;)

File: types/k6/experimental/websockets.d.ts
Comment (mstoykov) [inline types/k6/experimental/websockets.d.ts:96]: we don't support `blob` and likely never will - so maybe drop it :thinking:

**Review by mstoykov — APPROVED** [review]

File: types/k6/experimental/websockets.d.ts
Comment (oleiade) [reply types/k6/experimental/websockets.d.ts:96]: Fair enough 👍🏻

**Review by mstoykov — APPROVED** [review]

**Comment by mstoykov** [comment]
@oleiade maybe rename this to adding "k6 v0.40.0" ? It doesn't seem like we have any other API changes.

**Comment by oleiade** [comment]
Ready to merge

**Pull request merged**
